tests/bsim/bluetooth/host/l2cap/multilink_peripheral/src/central.c

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/att.h>
+#include <zephyr/bluetooth/l2cap.h>
+#include <zephyr/logging/log.h>
+
+#include "testlib/scan.h"
+#include "testlib/conn.h"
+
+#include "babblekit/flags.h"
+#include "babblekit/sync.h"
+#include "babblekit/testcase.h"
+
+/* local includes */
+#include "data.h"
+
+LOG_MODULE_REGISTER(central, CONFIG_APP_LOG_LEVEL);
+
+static int total_rx;
+static struct bt_l2cap_le_chan le_chan;
+
+void sent_cb(struct bt_l2cap_chan *chan)
+{
+	TEST_FAIL("Tester should not send data");
+}
+
+int recv_cb(struct bt_l2cap_chan *chan, struct net_buf *buf)
+{
+	LOG_DBG("received %d bytes", buf->len);
+	total_rx++;
+
+	return 0;
+}
+
+void l2cap_chan_connected_cb(struct bt_l2cap_chan *chan)
+{
+	LOG_DBG("%p", chan);
+}
+
+void l2cap_chan_disconnected_cb(struct bt_l2cap_chan *chan)
+{
+	LOG_DBG("%p", chan);
+}
+
+static struct bt_l2cap_chan_ops ops = {
+	.connected = l2cap_chan_connected_cb,
+	.disconnected = l2cap_chan_disconnected_cb,
+	.recv = recv_cb,
+	.sent = sent_cb,
+};
+
+int server_accept_cb(struct bt_conn *conn, struct bt_l2cap_server *server,
+		     struct bt_l2cap_chan **chan)
+{
+	memset(&le_chan, 0, sizeof(le_chan));
+	le_chan.chan.ops = &ops;
+	*chan = &le_chan.chan;
+
+	return 0;
+}
+
+static struct bt_l2cap_server test_l2cap_server = {.accept = server_accept_cb};
+
+static int l2cap_server_register(bt_security_t sec_level)
+{
+	test_l2cap_server.psm = 0;
```

Comment:
```suggestion
static int l2cap_server_register(bt_security_t sec_level)
{
	static struct bt_l2cap_server test_l2cap_server = {.accept = server_accept_cb};

	test_l2cap_server.psm = 0;
```

Optional suggestion to reduce scope of variable

Reply:
Same can be applied to most other callback structures to minimize the number of global variables

Reply:
I have a slight pref for not having `static` vars in function bodies. PTSD from ctest/cmocka that doesn't clear global memory between runs.
This doesn't apply here, so I'll apply your suggestion